src/internal/kopia/wrapper_test.go

```diff
@@ -277,6 +281,325 @@ func (suite *BasicKopiaIntegrationSuite) TestMaintenance_WrongUser_Force_Succeed
 	require.NoError(t, err, clues.ToCore(err))
 }
 
+// Test that failing to put the storage blob will skip updating the maintenance
+// manifest too. It's still possible to end up halfway updating the repo config
+// blobs as there's several of them, but at least this gives us something.
+func (suite *BasicKopiaIntegrationSuite) TestSetRetentionParameters_NoChangesOnFailure() {
+	t := suite.T()
+
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	k, err := openKopiaRepo(t, ctx)
+	require.NoError(t, err, clues.ToCore(err))
+
+	w := &Wrapper{k}
+
+	// Enable retention.
+	err = w.SetRetentionParameters(ctx, repository.Retention{
+		Mode:     ptr.To(repository.GovernanceRetention),
+		Duration: ptr.To(time.Hour * 48),
+		Extend:   ptr.To(true),
+	})
+	require.Error(t, err)
```

Comment:
Nit: standard pattern
```suggestion
	require.Error(t, err, clues.ToCore(err))
```